pyproject.toml

```diff
@@ -136,17 +136,26 @@ convention = "numpy"
 "galleries/users_explain/text/text_props.py" = ["E501"]
 
 [tool.mypy]
+ignore_missing_imports = true
+enable_incomplete_feature = [
+  "Unpack",
+]
 exclude = [
   ".*/matplotlib/(sphinxext|backends|testing/jpl_units)",
   ".*/mpl_toolkits",
   # tinypages is used for testing the sphinx ext,
   # stubtest will import and run, opening a figure if not excluded
   ".*/tinypages",
+  "galleries/",
+  "doc/",
+  "lib/matplotlib/backends/",
+  "lib/matplotlib/sphinxext",
+  "lib/matplotlib/testing/jpl_units",
+  "lib/mpl_toolkits/",
```

Comment:
these are included in the first two regex entry (which is a regex to allow the `mypy -m matplotlib` form which looks at installed version which is needed for stubtest.

Additionally, `tests` _should_ be included, while we do not fully type hint it, it provides a set of code that was not a large burden to add minimal types to but helps to check more _usage_ than stubs alone do.
I would not mind expanding this to galleries once type hints are more established (so not super soon), but there are complexities such as reused variables which change types, etc.

I'm not totally opposed to adding `galleries` and `docs` to this list for now, though when run as `mypy lib/matplotlib` they are not needed.